### DISCUSSION TITLE
scRNA intergration

### DISCUSSION THREAD

**Discussion by rmauron** [GreenleafLab/ArchR#1852] [Questions / Documentation]
Hi, 
Could you please show me the shape the Seurat object should have for the point "8.1: Cross-platform linkage of scATAC-seq cells with scRNA-seq cells". 
I tried to create a seurat object from my "filtered_feature_bc_matrix.h5" and take only the "Gene expression" to create my seurat object. But it doesn't work.
So if you could tell me how it should look I could maybe overcome this point.

Thanks in advance for your help!

**Comment by rcorces** [comment]
You can look at the code of `addGeneIntegrationMatrix()` and see what aspects of a Seurat object are used. I believe the primary requirements are:

1. Needs to have assay named "RNA" (`"RNA" %in% names(seuratRNA@assays)`)
2. Needs to have a column in `meta.data` named for whatever you pass to the `groupRNA` param (`seRNA@meta.data[,groupRNA]`)

**Reply by rmauron** [reply]
Hi, thanks for the reply. 
Now it seems to run but for some reasons it breaks saying "Error in sample.int(length(x), size, replace, prob) : 
  too few positive probabilities".
  
  I attach my log file: 
[ArchR-addGeneIntegrationMatrix-ba2116114a737-Date-2023-02-22_Time-11-02-20.log](https://github.com/GreenleafLab/ArchR/files/10802409/ArchR-addGeneIntegrationMatrix-ba2116114a737-Date-2023-02-22_Time-11-02-20.log)

I don't understand what it stands for. My Seurat object seRNA has meta.data and I gave the "groupRNA" different columns from it and I always get the same output.

I hope my issue is clear

**Reply by rcorces** [reply]
I'm afraid I dont have enough information to help you
